apps/condo/domains/resident/utils/serverSchema/findOrganizationsByAddress.js

```diff
@@ -11,112 +10,224 @@ const {
-        }]
-    }))
+    }
+
+    if (!receipts.length) receipts = null
+
+    const meters = await getOrganizationMeters(organization, addressKey, properties, { accountNumber })
+
+    return {
+        id: organization.id,
+        name: organization.name,
+        tin: organization.tin,
+        type: organization.type,
+        receipts,
+        meters,
+    }
 }
 
-async function getOrganizationsWithMeters (organizationIds, query = {}) {
-    const organizationsWithMeters = await Promise.all(organizationIds.map(async id => {
-        const [meter] = await itemsQuery('Meter', {
-            where: { organization: { id }, deletedAt: null, ...query },
-            first: 1,
+async function getOrganizationReceipts (billingContext, addressKey, query = {}) {
+    if (!billingContext) return []
+    const billingAccounts = await find('BillingAccount', {
+        context: { id: billingContext.id },
+        property: { addressKey, deletedAt: null },
+        deletedAt: null,
+        ...query,
+    })
+    let receipts = []
+
+    if (billingAccounts.length && get(billingContext, 'lastReport.period')) {
+        receipts = await find('BillingReceipt', {
+            context: { id: billingContext.id },
+            property: { addressKey, deletedAt: null },
+            deletedAt: null,
+            period: billingContext.lastReport.period,
+            account: { id_in: billingAccounts.map(({ id }) => id) },
         })
-        return meter ? id : null
-    }))
-    return new Set(organizationsWithMeters.filter(Boolean))
+
+        const billingAccountsNumbersIndex = Object.fromEntries(
+            billingAccounts.map(account => [account.id, account.number])
+        )
+        receipts = receipts.map(receipt => ({
+            category: receipt.category,
+            balance: receipt.toPay,
+            accountNumber: billingAccountsNumbersIndex[receipt.account],
+            routingNumber: receipt.recipient.bic,
+            bankAccount: receipt.recipient.bankAccount,
+            address: get(receipt, 'raw.address', null),
+        }))
+    }
+
+    return receipts
 }
 
-async function getOrganizationsWithAcquiring (organizationIds) {
+async function getOrganizationMeters (organization, addressKey, properties, query = {}) {
+    const meterResourceOwners = await find('MeterResourceOwner', {
+        organization: { id: organization.id },
+        addressKey,
+        deletedAt: null,
+    })
+
+    let meters = await find('Meter', {
+        organization: { id: organization.id },
+        deletedAt: null,
+        resource: { id_in: meterResourceOwners.map(owner => owner.resource) },
+        property: { addressKey, deletedAt: null },
+        ...query,
+    })
+
+    const property = properties.find(p => p.addressKey === addressKey)
+    const address = property ? property.address : null
+
+    if (meters.length){
+        const meterReadings = await find('MeterReading', {
```

Comment:
It's a lot of data. Automatic meters submit readings twice an hour.

That's ~9k submissions per year per Meter.